From 30db99963bc6c032962321c6cc1b95e09d0beb2f Mon Sep 17 00:00:00 2001
From: kas <kas@example.com>
Date: Fri, 23 Aug 2024 17:56:50 +0200
Subject: [PATCH] msg

---
 bitbake/lib/bb/build.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bitbake/lib/bb/build.py b/bitbake/lib/bb/build.py
index 44d08f5c55c..8c98e90d00a 100644
--- a/bitbake/lib/bb/build.py
+++ b/bitbake/lib/bb/build.py
@@ -279,10 +279,11 @@ def exec_func_python(func, d, runfile, cwd=None):
     """Execute a python BB 'function'"""

     code = _functionfmt.format(function=func)
     bb.utils.mkdirhier(os.path.dirname(runfile))
     with open(runfile, 'w') as script:
+        script.write("#!/usr/bin/env python3\n\n")
         bb.data.emit_func_python(func, script, d)

     if cwd:
         try:
             olddir = os.getcwd()
--
2.34.1

